### DISCUSSION TITLE
Using public property for something sensitive like change_password

### DISCUSSION THREAD

**Discussion by redjersey** [livewire/livewire#5040] [1. Help] [answered]
I read this important-notes on the official site:

https://laravel-livewire.com/docs/2.x/properties#important-notes

It said: 

> Data stored in public properties is made visible to the front-end JavaScript. Therefore, you SHOULD NOT store sensitive data in them.

So let's say I want to build a livewire form which allows the user to change his login password. So for fields there will be:
- enter your current password
- enter new password
- confirm new password

So in this case I should not create public properties for these fields? If they are not bind with the public properties, how to validate them?

Please advise

**Comment by Martleby** [comment, marked as answer]
I think in this case it would be fine because you're asking the user to enter these details and so they'd exist client side when the user enters them. 

What you wouldn't want to do is send the user's current password to the browser (for example pre-populating the current password field) as it would then be available without the user entering it and could then be "discovered" by hacker types.

**Comment by michael-rubel** [comment]
It's fine to accept and validate these values. The user sends it over the network anyway, Livewire or not. But if you continue working with the component on subsequent requests (without reloading) it's probably a good idea to nullify those properties after the method completed its job.

Public properties in Livewire live over the component lifecycle on subsequent requests, so if you didn't nullify the password, a potential hacker might access it from the frontend if he has access to the browser.

**Comment by joshhanley** [comment]
@redjersey yep, I agree with what both @Martleby  and @michael-rubel have said 🙂

**Comment by redjersey** [comment]
OK Thanks :)
